docker/Dockerfile

```diff
@@ -0,0 +1,161 @@
+FROM ubuntu:14.04
+
+SHELL ["/bin/bash", "-c"]
+ENV DEBIAN_FRONTEND noninteractive
+# ENV DEBIAN_FRONTEND newt
+ENV CLICOLOR=1
+ENV LSCOLORS=GxFxCxDxBxegedabagaced
+ENV GREP_OPTIONS='--color=auto'
+
+RUN \
+  echo America/Los_Angeles | sudo tee /etc/timezone && \
+  sudo dpkg-reconfigure --frontend noninteractive tzdata
+
+RUN \
+  apt-get update -qq -y && \
+  apt-get upgrade -qq -y
+
+RUN apt-get install -qq -y  \
+  software-properties-common \
+  vim \
+  curl \
+  wget \
+  gawk \
+  sed \
+  findutils \
+  bc \
+  less \
+  htop \
+  man \
+  unzip \
+  git \
+  lsb-release \
+  build-essential \
+  make \
+  python-all \
+  libssl-dev
+
+RUN apt-get install -qq -y \
+  openssh-server
+
+ENV HOME /root
+ENV NVM_DIR $HOME/.nvm
+ENV NODE_VERSION v4.3.0
+ENV YARN_VERSION 0.17.8
+
+RUN wget -qO- https://raw.githubusercontent.com/creationix/nvm/v0.31.1/install.sh | /bin/bash
+RUN source $NVM_DIR/nvm.sh; \
+  nvm install $NODE_VERSION && \
+  nvm alias default $NODE_VERSION && \
+  nvm use default
+
+RUN apt-get install -qq -y \
+  ca-certificates \
+  g++ \
+  gcc \
+  libc6-dev \
+  curl \
+  mercurial \
+  bzr \
+  git-core
+
+ENV GOLANG_VERSION 1.7.4
+ENV GOLANG_DOWNLOAD_URL https://storage.googleapis.com/golang/go$GOLANG_VERSION.linux-amd64.tar.gz
+
+RUN curl -s $GOLANG_DOWNLOAD_URL | tar -C /usr/local -xz
+
+ENV GOPATH /go
+ENV GOROOT /usr/local/go
+ENV PATH /usr/local/go/bin:/go/bin:/usr/local/bin:$PATH
+
+RUN go get -u -x -v -a github.com/gopherjs/gopherjs
+
+RUN \
+  source $NVM_DIR/nvm.sh; \
+  echo "nvm: " `nvm --version` && \
+  echo "node: " `node --version` && \
+  echo "npm: " `npm --version` && \
+  npm install -g \ 
+    gulp@"^3.9.1" \
+    bower@"^1.7.9" \
+    source-map-support
+
+RUN echo -e \
+'{'\
+'\n'\
+'  "allow_root": true'\
+'\n'\
+'}' \
+  >> $HOME/.bowerrc
+
+RUN cat $HOME/.bowerrc
+
+RUN \
+  source $NVM_DIR/nvm.sh; \
+  npm install -g \
+    bfs-buffer@"^0.1.7" \
```

Comment:
Did all of these come from the `package.json`?  Is it possible to `COPY` that in and install off of it, or otherwise auto-generate this?